### PR TITLE
Add FirstK aggregation

### PR DESCRIPTION
**Pull request by satrana42**
Add FirstK aggregation in codebase along the lines of LastK aggregation.

### CODE REVIEW

File: fennel/testing/execute_aggregation.py [hunk 1]
```diff
@@ -169,6 +170,43 @@ def get_val(self):
...
+        ret = []
+        for v in self.vals:
+            if v not in ret:
+                ret.append(v)
+            if len(ret) == self.k:
+                break
+        return list(ret[: self.k])
+
+    def get_val(self):
+        return list(reversed(self.vals[-self.k :]))
```
Comment (satrana42) [inline fennel/testing/execute_aggregation.py:207]: This has been copied as is from LastK. It should be `return list(self.vals[: self.k])`

File: fennel/testing/execute_aggregation.py [hunk 2]
```diff
@@ -169,6 +170,43 @@ def get_val(self):
...
+            self.vals.remove(val)
+        if not self.dedeup:
+            return list(self.vals[: self.k])
+
+        ret = []
+        for v in self.vals:
+            if v not in ret:
+                ret.append(v)
+            if len(ret) == self.k:
+                break
```
Comment (aditya-nambiar) [inline fennel/testing/execute_aggregation.py:203]: since you anyways have a cutoff below can we not use a list comprehension to simplify this ?

**Review by aditya-nambiar — APPROVED** [review]

**Comment by aditya-nambiar** [comment]
Do increment pyproject.toml and Changelog.md

File: fennel/testing/execute_aggregation.py (hunk 2 above)
Comment (satrana42) [reply fennel/testing/execute_aggregation.py:203]: This didn't work:
```
fennel/testing/execute_aggregation.py:198: in del_val_from_state
    ret = [v for v in self.vals if v not in ret]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <list_iterator object at 0x133fcc820>

>   ret = [v for v in self.vals if v not in ret]
E   NameError: cannot access free variable 'ret' where it is not associated with a value in enclosing scope

fennel/testing/execute_aggregation.py:198: NameError
```

**Pull request merged**